src/take.ts

```diff
@@ -1,21 +1,33 @@
-import { Operator } from './types';
+import { Operator, END } from './types';
 
 export function take<T>(n: number): Operator<T, T> {
   return source => (_, sink) => {
     let talkback: any;
+    let ended = false;
     let taken = 0;
 
     source(0, (t, d) => {
-      if (t === 0) talkback = d;
-      else if (t === 1) {
-        sink(1, d);
+      if (t === 0) {
+        talkback = d;
+        sink(0, (_: END) => {
+          ended = true;
+          talkback(_);
```

Comment:
Same here, we use `_` for explicitly ignored arguments. Just use `2` here

Reply:
changed